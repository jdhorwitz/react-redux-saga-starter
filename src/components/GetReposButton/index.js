import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {requestRepos} from '../../actions/repos';
import styles from './styles.scss';

const HARDCODED_ORG_NAME = 'github';

const GetReposButton = ({dispatch, fetching}) => {
  const handleOnClick = () => dispatch(requestRepos(HARDCODED_ORG_NAME));

  return (
    <button className={styles.button} disabled={fetching} onClick={!fetching && handleOnClick}>
      Get repos
    </button>
  );
};

GetReposButton.propTypes = {
  dispatch: PropTypes.func.isRequired,
  fetching: PropTypes.bool.isRequired
};

const mapStateToProps = (state) => ({
  fetching: state.repos.fetching
});

export {GetReposButton as GetReposButtonPureComponent};
export default connect(mapStateToProps)(GetReposButton);
