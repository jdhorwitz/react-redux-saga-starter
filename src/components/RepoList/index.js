import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

const RepoList = ({fetching, repos}) => {
  if (fetching) {
    return <p>Loading</p>;
  }

  if (!repos) {
    return <p>No results</p>;
  }

  return (
    <ul>
      {repos.map(({name, url}) =>
        <li key={name}>
          <a href={url}>
            {name}
          </a>
        </li>
      )}
    </ul>
  );
};

RepoList.propTypes = {
  fetching: PropTypes.bool.isRequired,
  repos: PropTypes.array
};

RepoList.defaultProps = {
  repos: null
};

const mapStateToProps = (state) => ({
  fetching: state.repos.fetching,
  repos: state.repos.repos
});

export {RepoList as RepoListPureComponent};
export default connect(mapStateToProps)(RepoList);
