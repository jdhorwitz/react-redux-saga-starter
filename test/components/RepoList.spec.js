import React from 'react';
import renderer from 'react-test-renderer';
import {RepoListPureComponent as RepoList} from '../../src/components/RepoList';

const mockRepos = [
  {
    name: 'firstrepo',
    url: 'https://github.com/orgname/firstrepo'
  },
  {
    name: 'secondrepo',
    url: 'https://github.com/orgname/secondrepo'
  }
];

describe('<RepoList/>', () => {
  it('renders "Loading" while fetching', () => {
    const tree = renderer.create(<RepoList fetching={false} repos={null} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders repo list with results when not fetching', () => {
    const tree = renderer.create(<RepoList fetching={false} repos={mockRepos} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
