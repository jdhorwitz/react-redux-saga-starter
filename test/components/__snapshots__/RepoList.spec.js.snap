// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoList/> renders "Loading" while fetching 1`] = `
<p>
  No results
</p>
`;

exports[`<RepoList/> renders repo list with results when not fetching 1`] = `
<ul>
  <li>
    <a
      href="https://github.com/orgname/firstrepo"
    >
      firstrepo
    </a>
  </li>
  <li>
    <a
      href="https://github.com/orgname/secondrepo"
    >
      secondrepo
    </a>
  </li>
</ul>
`;
